mongodb-storage/src/main/java/io/lumeer/mongodb/MongoDbStorage.java

```diff
@@ -382,26 +382,34 @@ public void dropAttribute(final String collectionName, final DataFilter filter,
    }
 
    @Override
-   public <T> void removeItemFromArray(final String collectionName, final DataFilter filter, final String attributeName, final T item) {
-      database.getCollection(collectionName).updateOne(filter.<Bson>get(), pull(attributeName, MongoUtils.isDataDocument(item) ? new Document((DataDocument) item) : item));
+   public <T> void removeItemFromArray(final String collectionName, final DataFilter filter, final String attributeName, final T item, final boolean multi) {
+      if (multi) {
+         database.getCollection(collectionName).updateMany(filter.<Bson>get(), pull(attributeName, MongoUtils.isDataDocument(item) ? new Document((DataDocument) item) : item));
+      } else {
+         database.getCollection(collectionName).updateOne(filter.<Bson>get(), pull(attributeName, MongoUtils.isDataDocument(item) ? new Document((DataDocument) item) : item));
+      }
    }
 
    @Override
-   public <T> void removeItemsFromArray(final String collectionName, final DataFilter filter, final String attributeName, final List<T> items) {
+   public <T> void removeItemsFromArray(final String collectionName, final DataFilter filter, final String attributeName, final List<T> items, final boolean multi) {
       if (items.isEmpty()) {
          return;
       }
       if (MongoUtils.isDataDocument(items.get(0))) {
          List<Document> docs = new ArrayList<>();
          items.forEach((i) -> docs.add(new Document((DataDocument) i)));
-         removeItemsFromArrayInternal(collectionName, filter, attributeName, docs);
+         removeItemsFromArrayInternal(collectionName, filter, attributeName, docs, multi);
          return;
       }
-      removeItemsFromArrayInternal(collectionName, filter, attributeName, items);
+      removeItemsFromArrayInternal(collectionName, filter, attributeName, items, multi);
    }
 
-   private <T> void removeItemsFromArrayInternal(final String collectionName, final DataFilter filter, final String attributeName, final List<T> items) {
-      database.getCollection(collectionName).updateOne(filter.<Bson>get(), pullAll(attributeName, items));
+   private <T> void removeItemsFromArrayInternal(final String collectionName, final DataFilter filter, final String attributeName, final List<T> items, final boolean multi) {
+      if (multi) {
+         database.getCollection(collectionName).updateMany(filter.<Bson>get(), pullAll(attributeName, items));
```

Comment:
Is this `multi` parameter really necessary? Why cannot we always use `updateMany` method?